models/location_helpers.go

```diff
@@ -358,6 +358,21 @@ func RemoveBackupLocation(q *reform.Querier, id string, mode RemoveMode) error {
 		return err
 	}
 
+	artifacts, err := FindArtifacts(q, &ArtifactFilters{LocationID: id})
+	if err != nil {
+		return err
+	}
+
+	if len(artifacts) != 0 && mode == RemoveRestrict {
+		return status.Errorf(codes.FailedPrecondition, "backup location with ID %q has artifacts.", id)
+	}
+
+	for _, a := range artifacts {
```

Comment:
can't we remove all artifacts at once?

Reply:
We could, but I cannot find this approach in our codebase, we always remove item-by-item...